Lib/idlelib/run.py

```diff
@@ -583,6 +583,9 @@ def runcode(self, code):
         else:
             flush_stdout()
 
+    def runcommand(self, code):
+        return self.runcode(code, user=False)
```

Comment:
Create a file with `print(dir())`.  Run from command line and `__file__` is listed.  Run from IDLE editor and __file__ is missing.  Change `False` to `True` and __file__ is listed again.  I have no idea how running the setup commands in `runscript.run_module_event` and `pyshell.ModifiedInterpreter.prepend_path` in a different dict affect the result of subsequently running a file.